pulp_container/app/registry_api.py

```diff
@@ -1008,18 +1008,24 @@ def put(self, request, path, pk=None):
             add_content_units = [str(manifest.pk)]
             remove_content_units = []
 
-        dispatched_task = dispatch(
+        immediate_task = dispatch(
             add_and_remove,
             exclusive_resources=[repository],
             kwargs={
                 "repository_pk": str(repository.pk),
                 "add_content_units": add_content_units,
                 "remove_content_units": remove_content_units,
             },
+            immediate=True,
+            deferred=False,
```

Comment:
The goal of this rework is to not be blocking (even for 3 s) in the api. So i would rather not...

Reply:
The nuance is in client behavior. Podman once being throttled on manifest upload does not retry solely the PUT call, unfortunately, but restarts whole image upload..

Reply:
But it will head all the blobs and find them nicely lined up, right?

Reply:
for most of them, yes. because the blob recompression  is not guaranteed to be reproducible it might lead to digest change, sometimes.

Reply:
Oh, damn!

Reply:
yeah.. https://docs.pulpproject.org/pulp_container/workflows/push.html#push-content-to-a-repository read the Warning

Reply:
Does this process invalidate signatures on manifests?

Reply:
before pushing to the registry you would get something like:
```
$ podman push pulp3-source-fedora36.puffy.example.com/ipanova/alpine
Getting image source signatures
Checking if image destination supports signatures
Error: Copying this image would require changing layer representation, which we cannot do: "Would invalidate signatures"
```
so the client would need to use options `--remove-signatures` and `--sign-by` during push operation.

Signature that land on the registry are valid because they referred to a certain digest, is manifest digest changes than new signature will be produced and pushed.